lib/compiler/passes/generate-bytecode.js

```diff
@@ -269,13 +275,14 @@ function generateBytecode(ast) {
     );
   }
 
-  function buildSemanticPredicate(code, negative, context) {
-    const functionIndex = addFunctionConst(Object.keys(context.env), code);
+  function buildSemanticPredicate(node, negative, context) {
+    const functionIndex = addFunctionConst(Object.keys(context.env), node.code);
 
     return buildSequence(
       [op.UPDATE_SAVED_POS],
       buildCall(functionIndex, 0, context.env, context.sp),
       buildCondition(
+        node.match | 0,
```

Comment:
why are you bitwise oring node.match with zero?  what does this mean?

Reply:
Not really, no.

Reply:
It turns out this is a substitute for `(if node.?match === 0)`

This is bad code and should not land.  We are not creating a trivia contest.

Reply:
Not really. `node.?match === 0` produces `true` or `false`, `node.match | 0` produces an integer, which represents ternary logic `<0`, `==0`, and `>0`.

Reply:
> Not really. node.?match === 0 produces true or false, node.match | 0 produces an integer, which represents ternary logic <0, ==0, and >0.

This is the kind of answer that would get the PR banned in almost any project

Reply:
It is clear that you don't even understand what other people think is the problem here